Deduplicate esfa subscriber lists

steventux opened the pull request:
https://trello.com/c/qqw5RbKb/556-change-this-email-signup-from-education-and-education-and-skills-funding-agency-to-education-and-skills-funding-agency

Two subscriber lists: 
`Education and Education and Skills Funding Agency` and 
`Education and Skills Funding Agency`
contain around 2000 subscriptions each with some subscribers on both lists.
We only want the latter so migrate subscriptions to this list and remove the `Education and Education and Skills Funding Agency` list.

binaryberry commented on db/migrate/20181019072511_deduplicate_esfa_subscriber_list.rb:
```diff
@@ -0,0 +1,30 @@
+
+      updated = scope.update_all(subscriber_list_id: new_list.id)
+
+      puts "Added #{updated} subscriptions to '#{new_list.title}'"
```
Did you want to display `updated` or `updated.count` here?

steventux replied:
@binaryberry `update_all` returns the number of affected records so this produced the output `Added 1528 subscriptions to 'Education and Skills Funding Agency'` on integration

steventux commented:
This ran successfully on integration:

```
14:24:05 == 20181019072511 DeduplicateEsfaSubscriberList: migrating ====================
14:24:05 ... 1821 subscriptions for 'Education and Education and Skills Funding Agency'
14:24:05 ... 2580 subscriptions for 'Education and Skills Funding Agency'
14:24:05 ... 289 subscribers exist in both lists
14:24:05 ... 1528 subscriptions to migrate from 'Education and Education and Skills Funding Agency' to 'Education and Skills Funding Agency'
14:24:06 ... Added 1528 subscriptions to 'Education and Skills Funding Agency'
14:24:07 ... Deleted 'Education and Education and Skills Funding Agency' with remaining subscriptions
14:24:07 == 20181019072511 DeduplicateEsfaSubscriberList: migrated (2.2129s) ===========

cbaines commented:
Could you use the rake task instead? [1] Looking at the approaches, it does look like there are differences, but it seems like it would work for this.

1: https://github.com/alphagov/email-alert-api/blob/master/lib/tasks/manage.rake#L83-L86

cbaines commented:
Could you use the rake task instead? [1] Looking at the approaches, it does look like there are differences, but it seems like it would work for this.

1: https://github.com/alphagov/email-alert-api/blob/master/lib/tasks/manage.rake#L83-L86

cbaines commented:
Could you use the rake task instead? [1] Looking at the approaches, it does look like there are differences, but it seems like it would work for this.

1: https://github.com/alphagov/email-alert-api/blob/master/lib/tasks/manage.rake#L83-L86

thomasleese commented:
@cbaines Good point! That's done now so I'll close this PR.

thomasleese commented:
@cbaines Good point! That's done now so I'll close this PR.

thomasleese commented:
@cbaines Good point! That's done now so I'll close this PR.

thomasleese commented:
@cbaines Good point! That's done now so I'll close this PR.

thomasleese commented:
@cbaines Good point! That's done now so I'll close this PR.
